pytorch_lightning/plugins/sync_batchnorm.py

```diff
@@ -19,7 +20,23 @@
 import pytorch_lightning as pl
 
 
-class SyncBatchNormPlugin:
+class LayerSync(ABC):
+    """Abstract base class for creating plugins that wrap layers of a model with synchronization logic
+    for multiprocessing.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def apply(model: "pl.LightningModule") -> "pl.LightningModule":
+        """Override this static method to apply synchronization to the layers of this model."""
+
+    @staticmethod
+    @abstractmethod
+    def revert(model: "pl.LightningModule") -> "pl.LightningModule":
+        """Override this static method to undo all modifications made in :meth:`apply`."""
+
+
```

Comment:
- i don't think these have to be static methods. someone could provide specialized instances. the strategy would still call self.layer_sync.apply/revert

- making these not nn.Module based means these could be applied to Lite
```suggestion
    @abstractmethod
    def apply(self, model: nn.Module) -> nn.Module:
        """Override this static method to apply synchronization to the layers of this model."""

    @abstractmethod
    def revert(self, nn.Module) -> nn.Module:
        """Override this static method to undo all modifications made in :meth:`apply`."""


```

Reply:
done